### PR TITLE
[bidi][java] Add filter auth handler

### PR DESCRIPTION
**Pull request by pujagani**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Add filter parameter while adding BiDi authentication handler. 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Added a new method `addAuthenticationHandler` with a URL filter parameter to the `Network` interface.
- Implemented the new method in `RemoteNetwork` to support URL-based filtering for authentication handlers.
- Refactored existing authentication handling logic to accommodate the new filtering mechanism.
- Added logging for handling malformed URLs during authentication.
- Added comprehensive tests for the new functionality, including multiple handlers and edge cases.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Network.java</strong><dd><code>Add URL filter parameter to authentication handler method</code></dd></summary>
<hr>

java/src/org/openqa/selenium/remote/Network.java

<li>Added a new method <code>addAuthenticationHandler</code> with a URL filter <br>parameter.<br> <li> Updated the interface to include the new method.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14349/files#diff-c60d5de1ba873d50b4fcb9e2e39e9db7bf792216fae033d7655ea8a9ee871d27">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>RemoteNetwork.java</strong><dd><code>Implement URL filtering for authentication handlers</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/remote/RemoteNetwork.java

<li>Implemented the new <code>addAuthenticationHandler</code> method with URL filter.<br> <li> Refactored authentication handling to support URL filtering.<br> <li> Added logging for malformed URLs.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14349/files#diff-d08d67a345515124ab64ce0074d8d9ecadda2b054a3c643042b4ed02deb08be0">+70/-23</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>WebNetworkTest.java</strong><dd><code>Add tests for authentication handler with URL filter</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/WebNetworkTest.java

<li>Added tests for the new <code>addAuthenticationHandler</code> method with URL <br>filter.<br> <li> Included tests for multiple handlers and edge cases.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14349/files#diff-246b58d71d0d6018c289289fa092c544b3b733a7bdeb8563f62eb286c401b68c">+70/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: java/src/org/openqa/selenium/remote/RemoteNetwork.java [hunk 1]
```diff
@@ -30,50 +36,90 @@
...
+
+  private void interceptAuthTraffic() {
+    this.network.addIntercept(new AddInterceptParameters(InterceptPhase.AUTH_REQUIRED));
+
+    network.onAuthRequired(
+        responseDetails -> {
+          String requestId = responseDetails.getRequest().getRequestId();
+
+          try {
+            URL url = new URL(responseDetails.getRequest().getUrl());
```
Comment (shs96c) [inline java/src/org/openqa/selenium/remote/RemoteNetwork.java]: Prefer to use `URI` rather than `URL` since `URL.equals` does a domain name lookup. You can create one with `URI.create`

**Review by shs96c — APPROVED** [review]
Approach LGTM. One comment, which you don't have to address.

File: java/src/org/openqa/selenium/remote/RemoteNetwork.java (hunk 1 above)
Comment (pujagani) [reply java/src/org/openqa/selenium/remote/RemoteNetwork.java]: Alright. Thank you!
